blockchains/eth/lib/next_interval_calculator.js

```diff
@@ -0,0 +1,47 @@
+const constants = require('./constants');
+
+/**
+ * Return the next interval to be fetched.
+ * NOTE: this method modifies the member variables of its argument
+ *
+ * @param {*} worker A worker object, member variables would be modified
+ * @returns An object like so:
+ * {
+ *  success: Boolean,
+ *  fromBlock: Integer,
+ *  toBlock: Integer
+ * }
+ */
+async function nextIntervalCalculator(worker) {
```

Comment:
I am externally modifying the member variables of a class here. Not sure if this is a good pattern but could not think of another way to reuse code in our current design. Ideas are welcome.